lib/streamlit/elements/widgets/data_editor.py

```diff
@@ -377,12 +377,12 @@ def _apply_dataframe_edits(
     if data_editor_state.get("edited_rows"):
         _apply_cell_edits(df, data_editor_state["edited_rows"], dataframe_schema)
 
-    if data_editor_state.get("added_rows"):
-        _apply_row_additions(df, data_editor_state["added_rows"], dataframe_schema)
-
     if data_editor_state.get("deleted_rows"):
         _apply_row_deletions(df, data_editor_state["deleted_rows"])
 
+    if data_editor_state.get("added_rows"):
```

Comment:
could you please add a comment why it is important that this comes after the deletion and that the sequence is important?

Reply:
Added 👍